### ISSUE TITLE
Make REQUESTED_SERVER_NAME DNS compliant 

### ISSUE THREAD

**Issue by kpanic9** [istio/istio#53426]
(This is used to request new product features, please visit <https://github.com/istio/istio/discussions> for questions on using Istio)

**Describe the feature request**
Envoy 1.32 has added additional validation for %REQUESTED_SERVER_NAME% filed in the Envoy access logs. Because Istio SNIs are not following DNS name convention, it fails validation and warning is added to the REQUESTED_SERVER_NAME.

REQUESTED_SERVER_NAME before Istio 1.23.2: outbound_.8080_._.{service-name}.{namespace}.svc.cluster.local
REQUESTED_SERVER_NAME after Istio 1.23.2: **invalid:**outbound_.8080_._.{service-name}.{namespace}.svc.cluster.local

Can you change the SNI format, so it will not log warning.

**Describe alternatives you've considered**
-

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by howardjohn** [comment]
Thats... very annoying. I feel like this is plausible an Envoy bug. Their TLS library accepts our SNI for TLS verification, routing, etc, but logging it is a step too far?

Ill have to dig up the issue, but its pretty ambiguous whether underscores are valid in SNI anyways. 

cc @keithmattix thoughts?

**Comment by kyessenov** [comment]
@howardjohn it's not the underscores, it's the pipe character, which is definitely illegal in SNI. This comes from upstream cluster naming.

**Comment by howardjohn** [comment]
There is no pipes, we translate the pipes to ._ which is how we get `outbound.8080_._.{service-name}.{namespace}.svc.cluster.local`

**Comment by kyessenov** [comment]
Ah, I thought it was the pipe. The CVE was for "bad characters in logging", but it does make sense to sanitize it early on and stop usage in SNI to begin with.

**Comment by howardjohn** [comment]
That would break Istio entirely, it would be much preferable for us to just allow logging these. There is not even much consensus that it is violating TLS RFCs (see https://github.com/istio/istio/issues/52004 for previous discussion)

**Comment by kyessenov** [comment]
SNI is meant to be a DNS name so it seems perfectly reasonable to enforce valid alphanumeric and ".-" characters.

I think you can probably replace SNI in Istio with "outbound.port-8080" or something like that and remain compliant.

**Comment by howardjohn** [comment]
Did you get a chance to look through https://github.com/istio/istio/issues/52004?

I don't agree with the assertion that valid SNIs == valid DNS domains; the RFCs are far more ambiguous than this on the definition of allowed SNIs. In practice, the only TLS library I have seen reject these is apparently Java's; even pretty strict ones like rustls accept these. That de-facto makes them legal.

Envoy, as well, has historically supported these, so dropping support would be a major breaking change. 

Additionally, its virtually impossible for Istio to stop using these, and this format is critical to Istio functioning.

**Comment by kyessenov** [comment]
```"HostName" contains the fully qualified DNS hostname of the server,
   as understood by the client.  The hostname is represented as a byte
   string using UTF-8 encoding [[UTF8](https://datatracker.ietf.org/doc/html/rfc4366#ref-UTF8)], without a trailing dot.```

https://datatracker.ietf.org/doc/html/rfc4366 from 2006.

**Comment by kyessenov** [comment]
It's interesting that SNI uses UTF-8 so using ASCII validation for it is probably over strict for SNI and should be relaxed.  cc @yavlasov

**Comment by yanavlasov** [comment]
I agree that it can be relaxed in Envoy. The problem was logging non UTF-8 sequences. These should be sanitized since they will break logging, but everything else can be left alone.

**Comment by howardjohn** [comment]
I opened https://github.com/envoyproxy/envoy/pull/36950
